scripts/js/src/console.js

```diff
@@ -21,41 +22,27 @@ function run(afn) {
     return runner;
 }
 
-function rand() {
-    return (Math.random() * 65536) | 0;
-}
-
 class PRuntimeApi {
     constructor(endpoint) {
         this.api = axios.create({
             baseURL: endpoint,
             headers: {
-                'Content-Type': 'application/json',
-            }
+                'Content-Type': 'application/octet-stream',
+            },
+            baseURL: endpoint,
+            responseType: 'arraybuffer',
         });
     }
-    async req(method, data={}) {
-        const r = await this.api.post('/' + method, {
-            input: data,
-            nonce: { id: rand() }
-        });
-        if (r.data.status === 'ok') {
-            return JSON.parse(r.data.payload);
-        } else {
-            throw new Error(`Got error response: ${r.data}`);
-        }
+    async req(method, data = undefined) {
+        const r = await this.api.post('/prpc/' + method, data);
+        return pruntime_rpc.PhactoryInfo.decode(r.data);
```

Comment:
Wrong place to decode the `PhactoryInfo`, should be in getInfo.
Why not use the `pruntime_rpc.PhactoryAPI.create` to construct a api client?